openlibrary/core/lending.py

```diff
@@ -436,8 +421,12 @@ def return_loan(self):
 
     def delete(self):
         loan = dict(self, returned_at=time.time())
+        user_key = self['user']
+        account = OpenLibraryAccount.get(key=user_key)
         if self.get("stored_at") == 'ia':
-            ia_lending_api.delete_loan(self['ocaid'], userkey2userid(self['user']))
+            ia_lending_api.delete_loan(self['ocaid'], userkey2userid(user_key))
+            if account.itemname:
+                ia_lending_api.delete_loan(self['ocaid'], account.itemname)
```

Comment:
I imagine we want both to run, one looks for a loan under ol:username, the other for @itemname.